client/client.go

```
@@ -0,0 +1,116 @@
+	flag.Parse()
+
+	http.HandleFunc("/get", handleGet)
+	http.HandleFunc("/set", handleSet)
+	http.HandleFunc("/post/set", handlePostSet)
+	http.HandleFunc("/post/get", handlePostGet)
+
+	log.Println("Start work at :8111...")
+	err := http.ListenAndServe(*httpAddr, nil)
+	if err != nil {
+		log.Fatal("Error happend", err.Error())
+	}
+}
+
+func handleSet(w http.ResponseWriter, r *http.Request) {
+	mainTmpl.Execute(w, "")
+}
+
+func defaultAPI() api.API {
+	return api.NewAPI(*urlAPI)
+}
+
+func handlePostSet(w http.ResponseWriter, r *http.Request) {
+	a := defaultAPI()
+	resp, err := a.Set(r.FormValue("phone"), r.FormValue("name"))
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		errResp, _ := json.Marshal("Error happend: " + err.Error())
+		w.Write(errResp)
+		return
+	}
+
+	makeResponse(w, r,resp)
+}
+
+func handleGet(w http.ResponseWriter, r *http.Request) {
+	retNameTmpl.Execute(w, "")
+}
+
+func handlePostGet(w http.ResponseWriter, r *http.Request) {
+	a := defaultAPI()
+
+	resp, err := a.Get(r.FormValue("phone"))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		errResp, _ := json.Marshal("Error happend: " + err.Error())
+		w.Write(errResp)
+		return
+	}
+
+	makeResponse(w, r,resp)
+}
+
+func checkUnmarshalError(w http.ResponseWriter, err error) bool {
+	if err != nil {
+		errResp, err := json.Marshal("Invalid response from server.\n" + err.Error())
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return false
+		}
+		w.Write(errResp)
+		return false
+	}
+	return true
+}
+
+func makeResponse(w http.ResponseWriter, r *http.Request, resp *http.Response) {
+	respBody, err := ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode != http.StatusOK {
+		w.WriteHeader(resp.StatusCode)
+		response := Errors{}
+
+		err = json.Unmarshal(respBody, &response)
+		if !checkUnmarshalError(w, err) {
+			return
+		}
+		errorTmpl.Execute(w, response.Errs)
+		return
+	}
+
+	w.WriteHeader(resp.StatusCode)
+	w.Write(respBody)
```

I assume here is error ignored
Please never ignore errors